build/Shared/HashCodeCombiner.cs

```diff
@@ -34,6 +34,12 @@ internal void AddObject(int i)
             AddHashCode(i);
         }
 
+        internal void AddObject(bool b)
+        {
+            CheckInitialized();
```

Comment:
Do we need to call this method here because the overloaded that accepts `int` parameter does that?

https://github.com/NuGet/NuGet.Client/blob/f6e5a71a4d1ac6fd4f461cfe0097c7a6f4496c2e/build/Shared/HashCodeCombiner.cs#L31-L35

Reply:
Yes. When calling this method with a bool it calls the overload accepting type object and boxes the bool as a Boolean.

Reply:
I am sorry if I was not clear in my message. My understanding is we can skip invoking `CheckInitialized();` method in `internal void AddObject(bool b)` method because the overloaded `internal void AddObject(int i) ` method does that. It is not a blocking comment though :)

Reply:
@kartheekp-ms Please let me know if you I'm not understanding correctly but I think you are saying that because the int overload calls CheckInitialized the bool overload does not need to? Please note the overload that accepts bool does not call into the overload that accepts int so we still need this overload to call CheckInitialized.

Reply:
@kartheekp-ms 
It looks `AddHashCode(int i)` here doesn't implement `CheckInitialized();` in it, so there is no double init is happening.

https://github.com/NuGet/NuGet.Client/blob/dd8eb8ae0e9e1c2aa3c079129791afd99f7c348d/build/Shared/HashCodeCombiner.cs#L26-L29

Reply:
@jebriede - I think you are correct. My bad somehow, I read the next line `AddHashCode(b ? 1 : 0);` as `AddObject(b ? 1 : 0);`.  Thanks for the clarification.